Sources/Core/Shared/GroupOperation.swift

```diff
@@ -185,22 +248,38 @@ public class GroupOperation: Operation, OperationQueueDelegate {
 
     /**
      The group operation acts as its own queue's delegate. When an operation is added to the queue,
-     assuming that the finishing operation has not started (or finished), and the operation is not
-     the finishing operation itself, then we add the operation as a dependency to the finishing
-     operation.
+     assuming that the finishing operation has not started (or finished), the operation is not
+     the finishing operation itself, and the CanFinishOperation has not yet started,
+     then we add the operation as a dependency to the CanFinishOperation.
```

Comment:
Hmm.

Arguably, the finishing operation is also private and an implementation detail of GroupOperation.

How about something like:

> The group operation acts as its own queue's delegate. When an operation is added to the queue, assuming that the GroupOperation is not yet finishing or finished, then we add the operation as a dependency to an internal operation.

> The purpose of this is to keep the internal operation as a "barrier" child operation that executes when there are no more operations in the group, safely handling the transition GroupOperation state.